### PR TITLE
Reactive RestEasy JAX-RS client - passing entities

### PR DESCRIPTION
**Pull request by michalszynkiewicz**

### CODE REVIEW

File: extensions/resteasy-reactive/quarkus-jaxrs-client/deployment/src/main/java/io/quarkus/resteasy/reactive/client/deployment/JaxrsClientProcessor.java [hunk 1]
```diff
@@ -241,12 +248,37 @@ public void registerInvocationCallbacks(CombinedIndexBuildItem index, ResteasyRe
...
 
-                    ResultHandle result = m
-                            .invokeInterfaceMethod(
-                                    MethodDescriptor.ofMethod(Invocation.Builder.class, "method", Object.class, String.class,
-                                            Class.class),
-                                    builder, m.load(method.getHttpMethod()), m.loadClass(method.getSimpleReturnType()));
-
+                    ResultHandle result;
+                    if (bodyParameterIdx != null) {
+                        String mediaTypeValue = "application/octet-stream";
```
Comment (stuartwdouglas) [inline extensions/resteasy-reactive/quarkus-jaxrs-client/deployment/src/main/java/io/quarkus/resteasy/reactive/client/deployment/JaxrsClientProcessor.java:253]: This should use the constant javax.ws.rs.core.MediaType#APPLICATION_OCTET_STREAM

File: extensions/resteasy-reactive/quarkus-jaxrs-client/deployment/src/main/java/io/quarkus/resteasy/reactive/client/deployment/JaxrsClientProcessor.java [hunk 2]
```diff
@@ -222,6 +226,9 @@ public void registerInvocationCallbacks(CombinedIndexBuildItem index, ResteasyRe
                                             String.class, Object[].class),
                                     tg, m.load(p.name), array);
                         }
+                        if (p.parameterType == ParameterType.BODY) {
```
Comment (stuartwdouglas) [inline extensions/resteasy-reactive/quarkus-jaxrs-client/deployment/src/main/java/io/quarkus/resteasy/reactive/client/deployment/JaxrsClientProcessor.java:229]: This should really be an 'else if', the end result is the same but it make the code easier to read.

**Review by stuartwdouglas — CHANGES_REQUESTED** [review]
Looks good, just a couple of minor things.

**Comment by michalszynkiewicz** [comment]
@stuartwdouglas thanks for the review, fixed

**Review by stuartwdouglas — APPROVED** [review]

**Pull request merged**